### DISCUSSION TITLE
Merge audio and video

### DISCUSSION THREAD

**Discussion by gregfr** [OvenMediaLabs/OvenMediaEngine#1126] [Help] [answered]
Greetings

I've noticed OME HLS output has separated audio and video tracks. Of course it sounds like a good idea, but I never see any other server do so. I was wondering if it would possible to merge the audio track on each video track, and if it would improve compatibility?

Regards

**Comment by getroot** [comment, marked as answer]
YouTube has already provided audio and video separately for a long time.
Even in HLS, providing Audio/Video in one file is an old specification. The latest HLS specification recommends serving each media as a separate file.

https://datatracker.ietf.org/doc/html/draft-pantos-hls-rfc8216bis

Serving media in one file has many drawbacks. For example, if you provide 3 audio quality and 3 video quality, you have to provide 9 files. And if there are 3 multilingual audios, there will be more. This means that a lot of duplicate data is accumulated in the cache and the hit rate is very low.

**Comment by gregfr** [comment]
I totally agree with you. However I was wondering if there is a setting to switch to the "merged" behavior? I'm having issues with the player's source switching so I'm exploring all directions.

**Comment by getroot** [comment]
Unfortunately, OvenMediaEngine does not provide a merged segment option. 
FYI it's for Low-Latency HLS, not a limitation of OvenMediaEngine. You should probably use a streaming server that supports Legacy HLS if you want that.

**Reply by naanlizard** [reply]
I'll jump in here because we use it

What are the plans for legacy hls? Even if unsupported, will it be left in ome for at least a year or two?

**Reply by getroot** [reply]
I'm not going to delete the source code of legacy HLS. But it's not tested, so it's not guaranteed to be stable (it might stop working due to other side effects). Also, new features (ABR, Multilingual Audio, AdmissionWebhooks, etc.) are not updated for it.
